salt/grains/core.py

```diff
@@ -2352,22 +2397,35 @@ def _hw_data(osdata):
             'manufacturer': 'sys_vendor',
             'biosreleasedate': 'bios_date',
             'uuid': 'product_uuid',
-            'serialnumber': 'product_serial'
+            'serialnumber': 'product_serial',
+            'mainboard_manufacturer': 'board_vendor',
+            'mainboard_name': 'board_name',
+            'mainboard_serial': 'board_serial',
+            'mainboard_version': 'board_version',
+            'chassis_manufacturer': 'chassis_vendor',
+            'chassis_serial': 'chassis_serial',
+            'chassis_type': 'chassis_type'
         }
         for key, fw_file in sysfs_firmware_info.items():
             contents_file = os.path.join('/sys/class/dmi/id', fw_file)
             if os.path.exists(contents_file):
                 try:
                     with salt.utils.files.fopen(contents_file, 'r') as ifile:
-                        grains[key] = salt.utils.stringutils.to_unicode(ifile.read().strip())
+                        grains[key] = _clean_value(key, salt.utils.stringutils.to_unicode(ifile.read().strip()))
                         if key == 'uuid':
                             grains['uuid'] = grains['uuid'].lower()
+                        elif key == 'chassis_type':
+                            try:
+                                grains['chassis_type'] = chassis_types[int(grains['chassis_type'])]
+                            except KeyError:
+                                grains['chassis_type'] = 'Unknown'
                 except (IOError, OSError) as err:
                     # PermissionError is new to Python 3, but corresponds to the EACESS and
                     # EPERM error numbers. Use those instead here for PY2 compatibility.
                     if err.errno == EACCES or err.errno == EPERM:
                         # Skip the grain if non-root user has no access to the file.
                         pass
+        grains = dict([(key, val) for key, val in grains.items() if val is not None])
```

Comment:
You can use a dict comprehension here, it'll run quicker. We no longer support Python 2.6 so dict comprehensions are OK. Also, using six.iteritems(grains) will make it more efficient on Python 2:

```python
grains = {key: val for key, val in six.iteritems(grains) if val is not None}
```